Archisinal.md

```diff
@@ -0,0 +1,310 @@
+- **Oleh KulyKovskyi**  - [linkedin.com/in/oleh-kulykovskyi-329523234/](http://www.linkedin.com/in/oleh-kulykovskyi-329523234/)
+
+
+## Development Status :open_book:
+
+- [Medium](https://medium.com/@archisinal)
+- [Twitter](https://twitter.com/archisinal)
+- [Instagram](https://www.instagram.com/archisinal.io/)
+
+## Development Roadmap :nut_and_bolt: ##
+
+### Overview
+
+The development roadmap for Archisinal includes multiple milestones to ensure the successful launch and operation of the platform.
+
+### **Overview**
+
+- **Total Estimated Duration:**
+    - MVP: 2.5 months
+    - Post-MVP: 9 months
+    - Post-Launch: 10 months
+- **Full-Time Equivalent (FTE): 6**
+- **Total Costs:**
+    - MVP: $60,000
+
+## MVP (brief discussion)
+
+### **Milestone 1: Smart Contract Development**
+
+- **Estimated duration:** 1 month
+- **FTE:** 3
+- **Costs:** 24,000 USD
+
+**Overview:**
+
+- Develop smart contracts for NFT minting, buying, selling, and royalty distribution
+- Testing and deployment on Astar testnet Shibuya
+
+| Number | Deliverable | Specification |
+| --- | --- | --- |
+| 0a. | License | MIT |
+| 0b. | Documentation | We will provide both inline documentation of the code and a basic tutorial that explains how users can interact with the developed smart contracts. |
+| 0c. | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit & integration tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an article that explains the development and features of the smart contracts, as well as any challenges faced and lessons learned. |
+| 1. | NFT Minting | Develop a smart contract functionality for NFT minting when creators upload their designs. |
+| 2. | NFT Transfer | Develop a smart contract functionality for NFT transfer when a sale happens, including royalty distribution. |
+| 3. | Creator’s Administering | Develop a smart contract functionality for creators, including managing their project NFTs data and tracking sales. |
+| 4. | Buyer’s Administering | Develop a smart contract functionality for buyers, including tracking their purchases and NFT holdings. |
+| 5. | Marketplace Administration | Develop a smart contract functionality for marketplace administrators, including the ability to delist inappropriate items. |
+| 6. | Auction | Develop smart contract functionality for auctioning NFTs, allowing users to place bids and manage their auctions. |
+| 7. | Test Network Deployment | Deploy the developed smart contracts on a test network (Shibuya) for evaluation and testing. |
+| 8. | Quality, Performance, and Security Considerations | Ensure that the developed smart contracts meet quality, performance (weight), and security requirements. |
```

Comment:
How do you plan to test these things?

Reply:
To ensure both quality and security requirements are met, we will utilize static analysis tools such as **Clippy** and **Semgrep**. The reports generated by these tools will be attached to pull requests (PRs). Further, we plan to integrate these tools with GitHub CI.

For performance requirements, we plan to implement a series of integration performance tests. These tests will primarily focus on gas limit assertions and will be executed against the Astar testnet to account for the weights of host functions. The aim of such tests is to identify and optimize problematic functions.

To address security requirements, we will develop a set of integration tests that specifically target security vulnerabilities and edge cases. These tests will cover fundamental vulnerabilities such as reentrancy and function visibility. Additionally, we will conduct a manual self-audit of our code, leveraging our expertise in ink! and Rust and adhering to best practices and smart contract requirements. While currently there are only Solidity audit tools available publicly, some of the rules can be also applicable to WASM smart contracts.

Reply:
Of course, here are our plans:
a) we're going to implement a set of Typescript integration tests and leverage [typechain-polkadot](https://github.com/Brushfam/typechain-polkadot) to enhance their readability and user-friendliness.
b) polkadot.js API provides [gasConsumed](https://github.com/polkadot-js/api/blob/edde226740341396bc59bf96ad38ea559357bb9d/packages/types/src/interfaces/contracts/types.ts#L74) field, which we intend to utilize for performance tracking.